lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -586,7 +588,7 @@ function collectAllStringsInDir(dir) {
       if (seenStrings.has(ctc.message)) {
         ctc.meaning = ctc.description;
         const seenId = seenStrings.get(ctc.message);
-        if (seenId) {
+        if (seenId && strings[seenId]) {
           if (!strings[seenId].meaning) {
```

Comment:
This line was failing (`strings[seenId]` undefined) on the `Name` string. No idea why.

Reply:
this sounds like we're now skipping a duplicate and the collision check when we should be looking into it :/ does it still happen if you remove the `&& strings[seenId]`? if so, seems like we should investigate.

Reply:
Candidly, I don't have time to do this and make our deadline of this week (would rather focus on better styling). If it's just a "fyi there is a duplicate" warning we are missing out on, seems a P4.

Reply:
OK to file an issue?

Reply:
> If it's just a "fyi there is a duplicate" warning we are missing out on, seems a P4.

Maybe I misunderstood the purpose here, but if there's a collision we don't know about, doesn't it like totally wreck the TC pipeline?

If I misunderstood and these warnings don't really have any consequences, then agreed P4, file an issue (if even) and carry on.

Reply:
I haven't any idea... @exterkamp ?

Reply:
collisions come from duplicate `message`s, in this case `'Name'` in `lighthouse-core/lib/i18n/i18n.js | columnName` and `lighthouse-treemap/app/src/util.js | name`. In those cases, we log a warning (and increment the collision count) because it's a sign that we might be wasting money translating the same string.

There's a chance that the strings' `description`s make them sufficiently different that they might be translated differently, so then we fall back to a `[message, description]` tuple to differentiate them (internally we copy `description` to `meaning`, but it's the same value). If _those_ tuples still collide then we throw an error because that will wreck the TC pipeline.

In this case this is a bug in `collect-strings`...we look for collisions across all strings so far (global `seenStrings`), but we then look up the collided-with string on the strings found only in this current directory (local `strings`).  Since the collision is between strings in `lighthouse-treemap/...` and `lighthouse-core/...` you get the `undefined` error here.

This is a collision but the `descriptions` of `Name` are different so it's not a fatal one. If you want @connorjclark you could land this to get your PR landed and I could do a follow up this afternoon/tomorrow to fix `collect-strings` to correctly look across root directories. Maybe add a TODO just to be crystal clear?

Reply:
> If those tuples still collide then we throw an error because that will wreck the TC pipeline.

OK glad my memory was somewhere in the ballpark here 😅 

SG thanks @brendankenny!

Reply:
thanks for the explanation. I'll add a TODO